x-pack/plugins/security_solution/cypress/tasks/api_calls/rules.ts

```diff
@@ -41,7 +45,7 @@ export const createCustomRule = (rule: CustomRule, ruleId = 'rule_testing', inte
       severity: rule.severity.toLocaleLowerCase(),
       type: 'query',
       from: 'now-50000h',
-      index: rule.index,
+      index: rule.dataSource.type === 'indexPatterns' ? rule.dataSource.index : '',
```

Comment:
I would follow up this in a different/follow up PR, adding the capability of creating also a rule with data view at first glance is more complex, since first, we need to get the data view id.